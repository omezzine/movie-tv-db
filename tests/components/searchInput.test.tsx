import { describe, expect, test, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import SearchInput from '../../src/components/searchInput';
import React from 'react';
import renderer from 'react-test-renderer';

describe('DB Card test', () => {
  test('Should match snapshot', () => {
    const search = renderer
      .create(<SearchInput defaultValue={{}} onInputChange={(e) => ({})} onTypeChange={(e) => ({})} />)
      .toJSON();
    expect(search).toMatchSnapshot();
  });

  test('Should render correctly', () => {
    const { container } = render(
      <SearchInput defaultValue={{}} onInputChange={(e) => ({})} onTypeChange={(e) => ({})} />,
    );
    expect(container.querySelector('input')).toBeDefined();
    expect(container.querySelector('select')).toBeDefined();
  });
  test('Should call fn when input change', () => {
    const onInputChange = vi.fn();
    const onTypeChange = vi.fn();
    const { container } = render(
      <SearchInput defaultValue={{}} onInputChange={onInputChange} onTypeChange={onTypeChange} />,
    );
    fireEvent.change(container.querySelector('input') as HTMLElement, { target: { value: 'matrix' } });
    expect(onInputChange).toHaveBeenCalledWith({ query: 'matrix', page: '1' });
  });
  test('Should call fn when select change', () => {
    const onInputChange = vi.fn();
    const onTypeChange = vi.fn();
    const { container } = render(
      <SearchInput defaultValue={{}} onInputChange={onInputChange} onTypeChange={onTypeChange} />,
    );
    fireEvent.change(container.querySelector('select') as HTMLElement, { target: { value: 'tv' } });
    expect(onTypeChange).toHaveBeenCalledWith({ type: 'tv', page: '1' });
  });
});
