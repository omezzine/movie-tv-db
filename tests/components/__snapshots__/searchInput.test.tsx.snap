// Vitest Snapshot v1

exports[`DB Card test > Should match snapshot 1`] = `
<div
  className="search-input"
>
  <div
    className="input-group"
  >
    <input
      aria-describedby="basic-addon2"
      className="form-control"
      onChange={[Function]}
      type="text"
    />
    <div
      className="input-group-prepend"
    >
      <select
        aria-label="Select Search Type"
        className="form-select"
        defaultValue="movie"
        onChange={[Function]}
      >
        <option
          value="movie"
        >
          Movie
        </option>
        <option
          value="tv"
        >
          Serie
        </option>
      </select>
    </div>
  </div>
</div>
`;
